### PR TITLE
CI: Migrate ROCm build test from FlexCI to GitHub Actions

### PR DESCRIPTION
**Pull request by kmaehashi**
ROCm build test does not require GPU, so let's run on GitHub Actions.

### CODE REVIEW

File: .github/workflows/pretest.yml [hunk 1]
```diff
@@ -58,3 +58,48 @@ jobs:
...
+        wget -qO - https://repo.radeon.com/rocm/rocm.gpg.key | sudo apt-key add -
+        echo "deb [arch=amd64] http://repo.radeon.com/rocm/apt/${ROCM_REPO}/ ubuntu main" | sudo tee /etc/apt/sources.list.d/rocm.list
+        sudo apt-get -qqy update
+        sudo apt-get -qqy install rocm-dev hipblas hipfft hipsparse rocsparse rocrand rocthrust rocsolver rocfft hipcub rocprim rccl
+
+        echo "ROCM_HOME=/opt/rocm" >> "${GITHUB_ENV}"
+        echo "PATH=${ROCM_HOME}/bin:${PATH}" >> "${GITHUB_ENV}"
+        echo "LD_LIBRARY_PATH=${ROCM_HOME}/lib:${LD_LIBRARY_PATH}" >> "${GITHUB_ENV}"
+
+        hipconfig
```
Comment (emcastillo) [inline .github/workflows/pretest.yml:90]: How about using an AMD provided rocm docker image?

Comment (kmaehashi) [reply .github/workflows/pretest.yml:90]: Raised another PR! Let's compare which one is faster 😄  https://github.com/cupy/cupy/pull/6031

Comment (kmaehashi) [reply .github/workflows/pretest.yml:90]: It seems this PR is slightly faster than #6034 

This PR: https://github.com/cupy/cupy/runs/4130093885
#6034: https://github.com/cupy/cupy/runs/4131011246

**Comment by kmaehashi** [comment]
Let's use #6034 instead!

**Pull request closed without merging**